dds/DCPS/security/AccessControlBuiltInImpl.cpp

```diff
@@ -987,10 +987,8 @@ ::CORBA::Boolean AccessControlBuiltInImpl::check_remote_datawriter_register_inst
   if (0 == reader) {
     return CommonUtilities::set_security_error(ex, -1, 0, "AccessControlBuiltInImpl::check_remote_datawriter_register_instance: Invalid Reader pointer");
   }
-  if (0 == key) {
-    return CommonUtilities::set_security_error(ex, -1, 0, "AccessControlBuiltInImpl::check_remote_datawriter_register_instance: Invalid Topic Key");
-  }
-
+  // Key could be null if we don't have complete type objects
+  ACE_UNUSED_ARG(key);
```

Comment:
I thought the DDS Security spec required a check of the key but I see your point.  Any security plugin implementation is free to require the key to be non-nill and only work with those remotes.

Reply:
Yea, I think I meant to ask you about this, but it slipped my mind. I was wondering if we should have a setting or requirement for complete TypeObjects to be available if instance checking is enabled.